test/integration/src/main/java/io/pravega/test/integration/compatibility/CompatibilityChecker.java

```diff
@@ -221,7 +232,95 @@ private void checkStreamDelete() {
         assertFalse(streamManager.checkStreamExists(scopeName, streamName));
     }
 
-    public static void main(String[] args) throws DeleteScopeFailedException {
+    /**
+     * This method attempts to validate abort functionality after creating a transaction.
+     * @throws TxnFailedException if unable to commit or write to the transaction.
+     */
+    @Test(timeout = 20000)
+    private void checkTransactionAbort() throws TxnFailedException {
+        String scopeName = "transaction-abort-test-scope";
+        String streamName = "transaction-abort-test-stream";
+        createScopeAndStream(scopeName, streamName);
+        @Cleanup
+        EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(scopeName, ClientConfig.builder().controllerURI(controllerURI).build());
+        @Cleanup
+        TransactionalEventStreamWriter<String> writerTxn = clientFactory.createTransactionalEventWriter(streamName, new UTF8StringSerializer(),
+                EventWriterConfig.builder().build());
+        // Begin a transaction
+        Transaction<String> txn = writerTxn.beginTxn();
+        assertNotNull(txn.getTxnId());
+        // Writing to the transaction
+        txn.writeEvent("event test");
+        // Checking and validating the Transaction status.
+        assertEquals(Transaction.Status.OPEN, txn.checkStatus());
+        // Aborting the transaction
+        txn.abort();
+        // It must fail if we are going to write to an aborted or aborting transaction.
+        assertThrows(TxnFailedException.class, () -> txn.writeEvent("event test"));
+        String readerGroupId = UUID.randomUUID().toString().replace("-", "");
```

Comment:
Nit: create a function to instantiate a random reader id (similar to what you did for creating scopes and streams), instead of doing this every time.